extensions/BugModal/web/bug_modal.js

```diff
@@ -1328,6 +1328,131 @@ $(function() {
+            if(!attachmentIds) return;
+
+            // Fetch all attachments for this bug. This one request will get data for
+            // most of the attachment links on the page.
+            $.get(`/rest/bug/${BUGZILLA.bug_id}/attachment`, { 
+                Bugzilla_api_token: BUGZILLA.api_token,
+                exclude_fields: 'data' // Remove the large Base64 encoded string of the attachment
+            })
+            .then(responseData => {
+                if(responseData['bugs'][BUGZILLA.bug_id]) {
+                    responseData['bugs'][BUGZILLA.bug_id].forEach(attachment => {
+                        attachmentData[attachment['id']] = attachment
+                    });
+                }
+            })
+            .then(() => {
+                // Individually fetch attachment data for each attachment id that wasn't found above.
+                // We do this because, although the bmo api supports fetching mulitple attachments by id,
+                // when an attachment id that the user is not permitted to see is included, the entire request
+                // fails with a 401 error (unlike the rest endpoint for bugs which simply filters out the bugs
+                // the user cannot see).
+                return $.when(
+                    attachmentIds
+                    .filter(id => !(id in attachmentData))
+                    .map(attachmentId => {
+                        return (
+                            $.get(`/rest/bug/attachment/${attachmentId}`, { 
+                                Bugzilla_api_token: BUGZILLA.api_token,
+                                exclude_fields: 'data' // Remove the large Base64 encoded string of the attachment
+                            })
+                            .then(responeData => {
+                                if(responeData['attachments'][attachmentId]) {
+                                    attachmentData[attachmentId] = responeData['attachments'][attachmentId];
+                                }
+                            })
+                            .catch(e => e) // Acknowledge and move on so our promise resolves.
+                        );
+                    })
+                );
+            })
+            .then(() => {
+                // Now we have all attachment data the user is able to see.
+                attachmentLinks.forEach(attachmentLink => {
+                    let data = attachmentData[attachmentLink.id];
+                    if(data) {
+                        attachmentLink.element.setAttribute("title",  data.summary);
+                        $(attachmentLink.element).tooltip({
+                            position: { my: "left top+8", at: "left bottom", collision: "flipfit" },
+                            show: { effect: 'none' },
+                            hide: { effect: 'none' }
+                        });
+                        // TODO if it's a patch append a link to the diff viewer.
```

Comment:
Help! The perl get_attachment_links code had access on the backend to check if the diff viewer is enabled. Can I replicate that on the frontend somehow (hmm I guess I could pass in an html attr when generating the link...) OR can we just assume that the diff viewer is always enabled for now?